test/unit/node_shared_test.rb

```diff
@@ -20,6 +20,32 @@ class NodeSharedTest < ActiveSupport::TestCase
     assert_equal 5, html.scan('notes-grid-test').length
     assert html.scan('<td class="title">').length > 1
   end
+ 
+  test 'that NodeShared can be used to convert short codes like [button:foo:https://google.com] into tables which list buttons' do
+    before = "Here are some notes in a table: \n\n[button:Press me:/questions] \n\n[button:Cancel:https://google.com]\n\n`[button:Cancel:https://google.com]` This shouldn't get recognized because it's in ` ticks.\n\nMake sense?"
+    html = NodeShared.button(before)
+    assert html
+    assert_equal 1, html.scan('<a class="btn btn-primary"').length
+    assert_equal 1, html.scan('href=').length
+    assert_equal 5, html.scan('button-grid-test').length
```

Comment:
this used to say `button-grid`, I changed it to `button-grid-test`

Reply:
Oh actually perhaps we'd remove this line entirely, since we aren't adding anything with that class in - Or, we could decide to add it in along with `btn btn-primary` up on this line:

https://github.com/publiclab/plots2/pull/5058/files#diff-a59c448107dbd2c805314c24ea66376cR16

I'll make a suggestion there accordingly! But we'd want to choose some unique classname we'd like to use anyways... let me see...

```suggestion
    assert_equal 5, html.scan('inline-button-shortcode').length
```